src/lib/jsonParseV2/parseRegistration.cpp

```diff
@@ -0,0 +1,332 @@
+    return false;
+  }
+
+  if (!jsonHttp.HasMember("url"))
+  {
+    *errorStringP = "/" + fieldName + "/ must contain a field /url/";
+    return false;
+  }
+
+  const rapidjson::Value& url = jsonHttp["url"];
+
+  if (!url.IsString())
+  {
+    *errorStringP = "/url/ field of /" + fieldName + "/ must be a JSON string";
+    return false;
+  }
+
+  httpP->url = url.GetString();
+
+  if (httpP->url == "")
+  {
+    *errorStringP = "/url/ field of /" + fieldName + "/ cannot be the empty string";
+    return false;
+  }
+
+  std::string  host;
+  int          port;
+  std::string  path;
+  std::string  protocol;
+
+  if (parseUrl(httpP->url, host, port, path, protocol) == false)
+  {
+    *errorStringP = "/url/ field of /" + fieldName + "/ is invalid";
+    return false;
+  }
+
+  return true;
+}
+
+
+
+/* ****************************************************************************
+*
+* providerParse -
+*/
+static bool providerParse(ConnectionInfo* ciP, ngsiv2::Provider* providerP, const rapidjson::Value& provider, std::string* errorStringP)
+{
+  bool b;
+
+  if (!provider.IsObject())
+  {
+    *errorStringP = "/provider/ must be a JSON object";
+    return false;
+  }
+
+  if (provider.ObjectEmpty())
+  {
+    *errorStringP = "/provider/ empty";
+    return false;
+  }
+
+  if (provider.HasMember("http"))
+  {
+    b = httpParse(&providerP->http, provider["http"], "http", errorStringP);
+    if (b == false)
+    {
+      return false;
+    }
+  }
+  else
+  {
+    *errorStringP = "/provider/ must contain a field /http/";
+    return false;
+  }
+
+  if (provider.HasMember("supportedForwardingMode"))
+  {
+    const rapidjson::Value& supportedForwardingModeValue = provider["supportedForwardingMode"];
+    std::string             supportedForwardingMode;
+
+    if (!supportedForwardingModeValue.IsString())
+    {
+      *errorStringP = "/supportedForwardingMode/ must be a JSON string";
+      return false;
+    }
+
+    supportedForwardingMode = supportedForwardingModeValue.GetString();
+
+    //
+    // Check Valid strings for supportedForwardingMode
+    //
+    if      (supportedForwardingMode == "none")   providerP->supportedForwardingMode = ngsiv2::ForwardNone;
+    else if (supportedForwardingMode == "query")  providerP->supportedForwardingMode = ngsiv2::ForwardQuery;
+    else if (supportedForwardingMode == "update") providerP->supportedForwardingMode = ngsiv2::ForwardUpdate;
+    else if (supportedForwardingMode == "all")    providerP->supportedForwardingMode = ngsiv2::ForwardAll;
+    else
+    {
+      *errorStringP = "invalid value of /supportedForwardingMode/";
+      return false;
+    }
+  }
+  else
+  {
+    providerP->supportedForwardingMode = ngsiv2::ForwardAll;
+  }
+
+  return true;
+}
+
+
+
+/* ****************************************************************************
+*
+* parseRegistration -
+*/
+std::string parseRegistration(ConnectionInfo* ciP, ngsiv2::Registration* regP)
```

Comment:
In the current implementation, parsing logic should look for legacyForwarding field and ensure its value is true. Otherwise, it should raise a Not Implemented error. To be covered also by a .test.

Suggested message for that Not Implement error descprtion: "Current version only support legacyForwarding. Please make it explicit in your registration creation request".

Reply:
Do we really want to do this?
The only supported value is `true` and if it is not present we return error?
I'm probably missing something here ... doesn't make any sense to me

Reply:
I understand your point but I think it is better to make it explictit for clients to avoid different behavious as we complete the funcionality  (somehow backward compability problem).

For instance, with the current version of the code, a client using the request without legacyForwarding will have different results now (i.e. a registration is created using forwarding in NGSIv1) and in the future (i.e. a registration is created using NGSIv2 forwarding). "Forcing" legacyFowarding we will avoid that problem.

Reply:
ok, I can accept that

Reply:
Fixed in 1a49367